lib/pure/ioselects/ioselectors_epoll.nim

```diff
@@ -141,7 +141,7 @@ template checkFd(s, f) =
   if f >= s.maxFD:
     raiseIOSelectorsError("Maximum number of descriptors is exhausted!")
 
-proc registerHandle*[T](s: Selector[T], fd: SocketHandle,
+proc registerHandle*[T](s: Selector[T], fd: int | SocketHandle,
```

Comment:
The point of the ``SocketHandle`` abstraction is type safety. ``int | SocketHandle`` is **not** typesafe.